tensorflow/core/lib/bfloat16/bfloat16.h

```diff
@@ -21,7 +21,7 @@ limitations under the License.
 
 #include "tensorflow/core/platform/byte_order.h"
 
-#ifdef __CUDACC__
+#if defined(__CUDACC__) || __HIPCC__
```

Comment:
@yxsamliu since it's in `core/lib` i think it's possible to add compiler/toolchain-specific macro check. but `__HIPCC__` may not be the best for both `hip/hcc` toolchain and `hip-clang` toolchain as from CI build log we can observe this PR indeed breaks the build.

I'm wondering if it's possible to add new builtin macro in `hip-clang` or HIP headers so we know we are compiling under `hip-clang` toolchain.

Reply:
There is a macro `__HIP__` which is only defined by hip-clang. I will replace `__HIPCC__` with that.